src/component/directive.ts

```diff
@@ -271,13 +271,39 @@ QWeb.addDirective({
     let tattClass = node.getAttribute("t-att-class");
     let styleAttr = node.getAttribute("style");
     let tattStyle = node.getAttribute("t-att-style");
+
+    const dataAttrs = [...node.attributes].filter(a => /^(t-attf?-)?data-.+$/.test(a.name)).sort();
+    if (dataAttrs.length) {
+      const dashToCamel = (str: string) => {
+        return str.toLowerCase()
+          .replace(/(\-[a-z])/g, val => val.toUpperCase().replace('-', ''));
+      };
+      ctx.addLine("let dataAttrMap = {};")
+      let attrName = "", attrValue = "";
+      for (const dataAttr of dataAttrs) {
+        if (dataAttr.name.startsWith("t-att-")) {
+          attrName = dashToCamel(dataAttr.name.slice(11));
+          let v = ctx.getValue(dataAttr.value);
+          attrValue = typeof v === "string" ? ctx.formatExpression(v) : `scope.${v.id}`;
```

Comment:
could this not be a string?

Reply:
I need to triple check, but I think this could be a QWebVar too

Reply:
You were right.